### ISSUE TITLE
Minor but consistent issue with manage deployment

### ISSUE THREAD

**Issue by aroute** [ibm-mas/ansible-devops#503]
This is a repeatable experience. Manage deployment always fails at this step. Repeating the step solves it. Perhaps the author of the db2 Role can look into this?

```
TASK [ibm.mas_devops.suite_db2_setup_for_manage : Copy the setupdb script into the db2u pod] *************************************
Monday 07 November 2022  14:24:56 +0000 (0:00:00.352)       0:27:42.740 ******* 
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "oc cp /tmp/setupdb.sh db2u/c-db2w-shared-db2u-0:/tmp/setupdb.sh", "delta": "0:00:21.230014", "end": "2022-11-07 14:25:18.209902", "msg": "non-zero return code", "rc": 1, "start": "2022-11-07 14:24:56.979888", "stderr": "Defaulted container \"db2u\" out of: db2u, init-labels (init), init-kernel (init)\nerror: unable to upgrade connection: container not found (\"db2u\")", "stderr_lines": ["Defaulted container \"db2u\" out of: db2u, init-labels (init), init-kernel (init)", "error: unable to upgrade connection: container not found (\"db2u\")"], "stdout": "", "stdout_lines": []}
```

**Comment by thomas-mattsson** [comment]
@pmvester getting this consistently as well, fyi @jvc0107

**Comment by neilpatt** [comment]
I am also seeing a similar error running the ansible from a cli container. running 12.7.3

fatal: [localhost]: FAILED! => {"changed": true, "cmd": "oc cp /tmp/setupdb.sh db2u/c-db2w-shared-db2u-0:/tmp/setupdb.sh", "delta": "0:00:09.560918", "end": "2023-01-20 17:52:34.900960", "msg": "non-zero return code", "rc": 1, "start": "2023-01-20 17:52:25.340042", "stderr": "Defaulted container \"db2u\" out of: db2u, init-labels (init), init-kernel (init)\nerror: unable to upgrade connection: container not found (\"db2u\")", "stderr_lines": ["Defaulted container \"db2u\" out of: db2u, init-labels (init), init-kernel (init)", "error: unable to upgrade connection: container not found (\"db2u\")"], "stdout": "", "stdout_lines": []}

**Comment by durera** [comment]
Not sure what we can do about this one tbh ...

The code already waits until the Pod is ready: https://github.com/ibm-mas/ansible-devops/blob/master/ibm/mas_devops/roles/suite_db2_setup_for_manage/tasks/main.yml#L37-L50

```
- name: "Lookup db2 Pod"
  kubernetes.core.k8s_info:
    kind: Pod
    namespace: "{{ db2_namespace }}"
    label_selectors:
      - type=engine
      - app={{ db2_instance_name }}
    wait: yes
    wait_sleep: 30
    wait_timeout: 300 # 5 mins until we give up waiting for the pod to get into the expected state
    wait_condition:
      type: Ready
      status: "True"
  register: db2_pod
```

After this we should be able to reliably exec into the Pod.  We're not hitting this in our daily FVT environments ... which are deploying MAS with Db2 half a dozen times a day.

Do you know what is happening in the cluster/with the pod at the time you're seeing this?

I can add a dumb retry to the step so that it will at least automatically retry a few times before failing, that's not a very satisfying solution as it's not really addressing the issue of why the pod reports Ready status, and a few second later we're unable to work with one of the containers in the pod only seconds later.  However, if retying is generally a workable solution to this, we may as well have the code do that itself so you don't have to.

**Comment by durera** [comment]
Automatica retries are available in https://github.com/ibm-mas/ansible-devops/releases/tag/12.10.3 ... this release also contains a fix for #604 which may well be the underlying cause of this in some clusters
